CMakeLists.txt

```diff
@@ -0,0 +1,198 @@
+    # C4244 is a possible loss of data warning for integer conversions.
+    /w44244
+    # C4267 is a possible loss of data warning when converting from `size_t`.
+    /w44267)
+endif ()
+
+# WINDOWS CONFIGURATION.
+########################
+if (WIN32)
+  # COFF/PE and friends are somewhat limited in the number of sections they
+  # allow for an object file. We use this to avoid those problems.
+  add_compile_options(/bigobj /vd2)
+
+  # Fix Warning C4530: C++ exception handler used, but unwind semantics are not
+  # enabled.
+  add_compile_options(/EHsc)
+
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+  # Enable multi-threaded compilation for `cl.exe`.
+  add_compile_options(/MP)
+
+  # Force use of Unicode C and C++ Windows APIs.
+  add_definitions(-DUNICODE -D_UNICODE -DNOGDI -DNOMINMAX -DGOOGLE_GLOG_DLL_DECL=)
+
+  # Convenience flags to simplify Windows support in C++ source; used to
+  # `#ifdef` out some platform-specific parts of Mesos.  We choose to define
+  # a new flag rather than using an existing flag (`_WIN32`) because we want
+  # to give the build system fine-grained control over what code is #ifdef'd
+  # out in the future.  Using only flags defined by our build system to control
+  # this logic is the clearest and most stable way of accomplishing this.
+  add_definitions(-D__WINDOWS__)
+endif ()
+
+add_subdirectory(dockercfg)
+add_subdirectory(journald)
+add_subdirectory(logsink)
+
+# FIXME: due to import path in 'overlay/messages.proto'
+# NOTE: according to [1], overlay module is not required for Windows
+# [1] https://github.com/mesosphere/dcos-enterprise/tree/master/packages/mesos/extra/windows-ee.patches
+#add_subdirectory(overlay)
+
+
+# FIXME: due to limitation of protobuf generation process
+# and include in source
+include_directories(
+  .
+  build
+  "${BOOST_INCL}"
+  "${CURL_INCL}"
+  "${GLOG_INCL}"
+  "${MESOS_ROOT}/build/include"
+  "${MESOS_ROOT}/include"
+  "${PICOJSON_INCL}"
+  "${PROCESS_INCL}"
+  "${PROTOBUF_INCL}"
+  "${RAPIDJSON_INCL}"
+  "${STOUT_INCL}"
+)
+
+add_library(metrics SHARED metrics/isolator.cpp)
+
+set(CMAKE_CURRENT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/metrics)
+
+protobuf_generate(
+  LANGUAGE cpp
+  TARGET metrics
+  PROTOS metrics/messages.proto
+  APPEND_PATH ./metrics
+)
+
+target_link_libraries(metrics PRIVATE
+  ${OPENSSL_LIBRARIES}
+  glog
+  http_parser
+  mesos
+  mesos_protobufs
+  process
+  protobuf
+  zlib
+  $<$<PLATFORM_ID:Windows>:Mswsock>
+  $<$<PLATFORM_ID:Windows>:Userenv>
+  $<$<PLATFORM_ID:Windows>:ws2_32>
+)
+
+if (BUILD_TESTING)
+  add_subdirectory(tests)
+
+  # Test binary for metrics
+  add_executable(test-metrics tests/metrics_tests.cpp)
+
+  add_definitions(
+    -DMODULES_BUILD_DIR="${PROJECT_BINARY_DIR}"
+    -DSOURCE_DIR="${MESOS_ROOT}"
+    -DBUILD_DIR="${MESOS_ROOT}/build"
+    -DPKGMODULEDIR="nowhere"
+    -DLIBDIR="nowhere"
+    -DPKGLIBEXECDIR="nowhere"
+    -DTESTLIBEXECDIR="nowhere"
+    -DSBINDIR="nowhere"
+    -DPKGDATADIR="nowhere"
```

Comment:
Why do we need to define these at all? 

Is it because something (what exactly?) depends on these definitions existing but doesn't actually use a value?
Or is it just because something will break in a non-transparent way?

In any case, I would ask you to leave a comment explaining the need for these definitions, so that readers don't get lost in the future.

Reply:
https://github.com/dcos/dcos-mesos-modules/blob/master/Makefile.am#L192-L198

Reply:
done